codalab/worker/bundle_manager.py

```diff
@@ -347,11 +366,12 @@ def _filter_and_sort_workers(self, workers_list, bundle):
                               bundle.dependencies))
         def get_sort_key(worker):
             deps = set(worker['dependencies'])
+            worker_id = worker['worker_id']
 
             # if the bundle doesn't request GPUs (only request CPUs), prioritize workers that don't have GPUs
             if not self._compute_request_gpus(bundle):
-                return (-worker['gpus'], len(needed_deps & deps), worker['slots'] - len(worker['run_uuids']), random.random())
-            return (len(needed_deps & deps), worker['slots'] - len(worker['run_uuids']), random.random())
+                return (not has_gpu[worker_id], len(needed_deps & deps), worker['cpus'], random.random())
```

Comment:
Can we avoid having duplicate code here for the two cases?  Seems like defining a variable to be `not self._compute_request_gpus(bundle) and not has_gpu[worker_id]` (check my logic) and using that as the first variable would do the trick.